### PR TITLE
Change definition of `StatInfo` and add some lemmas

### PR DESCRIPTION
**Pull request by LorenzoLuccioli**
- Modify the def of `StatInfo` so that it is the difference of two `bayesBinaryRisk`.
- Add some API lemmas.
- Add `ENNReal.min_mul` and `ENNReal.mul_min`.
- Add `bayesBinaryRisk_ne_top`.
- Rename `bayesBinaryRisk_eq_integral_min` to `bayesBinaryRisk_eq_lintegral_min`.
- Add `bayesBinaryRisk_eq_lintegral_ennnorm`.
- Cleanup.

### CODE REVIEW

File: TestingLowerBounds/Divergences/StatInfo.lean [hunk 1]
```diff
@@ -41,15 +41,99 @@ variable {𝒳 𝒳' : Type*} {m𝒳 : MeasurableSpace 𝒳} {m𝒳' : Measurabl
...
+
+lemma statInfo_eq_min_sub_lintegral (μ ν : Measure 𝒳) [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    (π : Measure Bool) [IsFiniteMeasure π] :
+    statInfo μ ν π = min (π {false} * μ univ) (π {true} * ν univ)
+      - ∫⁻ x, min (π {false} * μ.rnDeriv (π ∘ₘ twoHypKernel μ ν) x)
+      (π {true} * ν.rnDeriv (π ∘ₘ twoHypKernel μ ν) x) ∂(π ∘ₘ twoHypKernel μ ν) := by
+  rw [statInfo_eq_min_sub, bayesBinaryRisk_eq_lintegral_min]
+
+lemma statInfo_eq_min_sub_lintegral' {μ ν ζ : Measure 𝒳} [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    [SigmaFinite ζ] (π : Measure Bool) [IsFiniteMeasure π]  (hμζ : μ ≪ ζ) (hνζ : ν ≪ ζ) :
```
Comment (RemyDegenne) [inline TestingLowerBounds/Divergences/StatInfo.lean]:
```suggestion
    [SigmaFinite ζ] (π : Measure Bool) [IsFiniteMeasure π] (hμζ : μ ≪ ζ) (hνζ : ν ≪ ζ) :
```

File: TestingLowerBounds/Divergences/StatInfo.lean [hunk 2]
```diff
@@ -41,15 +41,99 @@ variable {𝒳 𝒳' : Type*} {m𝒳 : MeasurableSpace 𝒳} {m𝒳' : Measurabl
...
+    (π : Measure Bool) [IsFiniteMeasure π] :
+    (statInfo μ ν π).toReal = min (π {false} * μ univ).toReal (π {true} * ν univ).toReal
+      - ∫ x, min (π {false} * μ.rnDeriv (π ∘ₘ twoHypKernel μ ν) x).toReal
+      (π {true} * ν.rnDeriv (π ∘ₘ twoHypKernel μ ν) x).toReal ∂(π ∘ₘ twoHypKernel μ ν) := by
+  have hμ : π {false} * μ univ ≠ ⊤ := ENNReal.mul_ne_top (measure_ne_top π _) (measure_ne_top μ _)
+  have hν : π {true} * ν univ ≠ ⊤ := ENNReal.mul_ne_top (measure_ne_top π _) (measure_ne_top ν _)
+  rw [statInfo_eq_min_sub, ENNReal.toReal_sub_of_le (bayesBinaryRisk_le_min μ ν π)]
+  swap; · simp only [ne_eq, min_eq_top, hμ, hν, and_self, not_false_eq_true]
+  rw [toReal_bayesBinaryRisk_eq_integral_min,
+    MonotoneOn.map_min (fun _ _ _ hb hab ↦ ENNReal.toReal_mono hb hab) hμ hν]
```
Comment (RemyDegenne) [inline TestingLowerBounds/Divergences/StatInfo.lean:137]:
```suggestion
    MonotoneOn.map_min (fun _ _ _ hb hab ↦ ENNReal.toReal_mono hb hab) hμ hν]

```

**Review by RemyDegenne — APPROVED** [review]

**Comment by RemyDegenne** [comment]
It looks like `ENNReal.mul_min` is not defined.

**Comment by LorenzoLuccioli** [comment]
Sorry I forgot to add the changes to `Binary.lean` and `EReal.lean` to the PR, now it should work.

File: TestingLowerBounds/Divergences/StatInfo.lean [hunk 3]
```diff
@@ -41,15 +41,100 @@ variable {𝒳 𝒳' : Type*} {m𝒳 : MeasurableSpace 𝒳} {m𝒳' : Measurabl
...
+
+lemma statInfo_eq_min_sub_lintegral (μ ν : Measure 𝒳) [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    (π : Measure Bool) [IsFiniteMeasure π] :
+    statInfo μ ν π = min (π {false} * μ univ) (π {true} * ν univ)
+      - ∫⁻ x, min (π {false} * μ.rnDeriv (π ∘ₘ twoHypKernel μ ν) x)
+      (π {true} * ν.rnDeriv (π ∘ₘ twoHypKernel μ ν) x) ∂(π ∘ₘ twoHypKernel μ ν) := by
+  rw [statInfo_eq_min_sub, bayesBinaryRisk_eq_lintegral_min]
+
+lemma statInfo_eq_min_sub_lintegral' {μ ν ζ : Measure 𝒳} [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    [SigmaFinite ζ] (π : Measure Bool) [IsFiniteMeasure π]  (hμζ : μ ≪ ζ) (hνζ : ν ≪ ζ) :
```
Comment (RemyDegenne) [inline TestingLowerBounds/Divergences/StatInfo.lean]:
```suggestion
    [SigmaFinite ζ] (π : Measure Bool) [IsFiniteMeasure π] (hμζ : μ ≪ ζ) (hνζ : ν ≪ ζ) :
```

**Review by RemyDegenne — APPROVED** [review]

**Pull request merged**